### PR TITLE
Improved array literal construction

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by LaszloLango** [comment]
LGTM

File: jerry-core/vm/vm.c [hunk 1]
```diff
@@ -1051,16 +1038,23 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
...
+              JERRY_ASSERT (ecma_is_value_undefined (ecma_get_named_data_property_value (prop_p)));
 
-              ecma_deref_ecma_string (index_str_p);
+              ecma_set_named_data_property_value (prop_p, stack_top_p[i]);
 
-              ecma_free_value (stack_top_p[i]);
+              /* The reference is moved so no need to free stack_top_p[i] except for objects. */
+              if (ecma_is_value_object (stack_top_p[i]))
+              {
+                ecma_free_value (stack_top_p[i]);
```
Comment (akiss77) [inline jerry-core/vm/vm.c]: Isn't this equivalent to `ecma_free_value_if_not_object`?

Comment (zherczeg) [inline jerry-core/vm/vm.c]: It is completely the opposite. Free if object.

Comment (akiss77) [inline jerry-core/vm/vm.c]: Ah my bad, mixed it up.

**Comment by akiss77** [comment]
LGTM

**Pull request merged**